### PR TITLE
Tidy up Great Expectations plugin code

### PR DESCRIPTION
**Pull request by samhita-alla**
Signed-off-by: Samhita Alla <aallasamhita@gmail.com>

# TL;DR
This PR cleans up the unused plugin code, removes repetitive code chunks, adds a new test case for local files, and fixes FlyteSchema usage in the tests.

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by wild-endeavor — DISMISSED** [review]
looks okay, bit of a rubber stamp.  calling transformers directly out of the type engine is something i don't fully get.

**Review by wild-endeavor — APPROVED** [review]
